protostar/cairo_testing/execution_environments/cairo_execution_environment.py

```diff
@@ -28,6 +31,12 @@ async def run_cairo_function(
         *args: Any,
         **kwargs: Any,
     ):
+        CheatableSysCallHandler.block_info_controller_state = (
+            self.state.block_info_controller_state
+        )
+        CheatableSysCallHandler.contracts_controller_state = (
+            self.state.contracts_controller_state
+        )
```

Comment:
Setting a static class property with a state instance is a big no-no, it was implemented before and was marked for removal in future versions. We'd like to avoid this.

Reply:
Agreed, there is no obvious reason to do that, though there are some subtle, but important ones. First, it is much easier to follow and explore the codebase without monkey patching, it is always somewhat confisuing to reason about state, you can't never be sure if there is no race condition. Second reason is we are close to rust migration, and Rust does not allow such hacks. I would prefer to not encourage any form of dependence on them, to simplify the integration in the future. Hope it makes sense.